### DISCUSSION TITLE
PointerPressed event is not always called in a custom control | Custom Control | Events

### DISCUSSION THREAD

**Discussion by BoTech-Development** [AvaloniaUI/Avalonia#17227] [Q&A]
### Describe the bug

- First of all it is important to know that this bug or question is related to one of my Repos: [ISim](www.github.com/BoTech-Development/ISim)
- I have programmed a new Custom Control which will be used to display schematic and it can be used to edit the schematic. [PaintControl.cs](https://github.com/BoTech-Development/ISim/blob/master/ISim/ViewModels/SchematicEditor/PaintControl.cs)
	- I have used this [Example](https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/Drawing/RectPainter) from Avalonia to get a idea how to program it.
- Furthermore I want to complete the new Feature ( in v1.0.11) that users can select Items via a rectangle on the surface.
- Due to this feature the method [PointerPressed](https://github.com/BoTech-Development/ISim/blob/ca95faeef51d43a0fb48b25aac309597e69c5ed1/ISim/ViewModels/SchematicEditor/PaintControl.cs#L220) has to be called every time when the user clicks on the surface.
	- Unfortunately this is not the case.
	- My solution was to call this method from the PaintControl itself and from the Window where the PaintControl is placed in.[Second call of the Event](https://github.com/BoTech-Development/ISim/blob/master/ISim/Views/SchematicEditorWindow.axaml.cs)
	- But causes a bigger problem: 
		- This method is sometimes called twice, once or not at all by a mouse click
**My Bug/Question: How can I program a custom control that has mouse events that are called at the right time.**

### To Reproduce

- To reproduce please build my program and navigate to the test button to open the schematic editor.

### Expected behavior

- My expected behavior was that the Pointer Events of an Custom Control gets called a n the correct time.

### Avalonia version

11.1.3

### OS

Windows

### Additional context

**I'm not entirely sure if the Avalonia version is correct, as I haven't worked on ISim for a while and I don't have a device available with which I could check this.**

**Comment by stevemonaco** [comment]
I'm pretty confident that this specific scenario works as intended and isn't an issue. At least there's far too much to sift through (a whole app). Be mindful for hit testing, you need to have a `Background` that is not `null` (the default). Once hit testing succeeds, input events are allowed.

Alternatively, you can implement `ICustomHitTest`:

```cs
public class PaintControl : Control, ICustomHitTest
{
    public bool HitTest(Point point) => true;
    ...
}
```

If this doesn't work, please create a minimal repro.
